### DISCUSSION TITLE
Some minor ioSender thoughts  and suggestions

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/ioSender#210] [Ideas]
I really like the new XL version.   After writing "A brief tour", some things stood out.  These could be seen as nit picking but my intent is not to annoy.

Why are there 2 3D View tabs?  They seem redundant. I guess the top 3D View is larger?  Seems like a point of confusion.  Drop the top one?

The first time ioSender is run it presents the dialog "Config file not found or invalid". This seems to scare people.  Drop "or invalid"?  Even better, why not just say "Creating config file"/OK?  Is there ever a reason to say No?

In Settings: Grbl.  There is a large, empty light gray rectangle.  I don't think it is ever used.  Is there a purpose for it?

In the XL version, could we swap the Coolant and Outline groups?  On a smaller screen, outline doesn't show.  I have found Outline to be incredibly useful for beginning CNCers - we make them run it and about half of them have realized they did something wrong before making firewood.  Having it on the screen will remind them to use it.

I continue to think having a very simple Probe Z  tab would be useful.  I have no numbers to quote but would not be surprised if the vast majority of probing is only Z finding.  Certainly after a manual tool change all that is needed is to find Z. I think the probing tabs are intimidating to beginners.  Heck, I consider myself pretty experienced and find them a bit that way. Stefan's videos go a long way towards making it easier to understand, though.

I think it would be useful to have a list of error and alarm codes available from the app.  Perhaps with some additional info to help the user resolve the problem. Maybe in Help?  

Same thing for a list of keyboard short cuts.  Is the wiki usage tip list up to date?

On the subject of keyboard short cuts.  Making them user assignable would be awesome. I realize that is a bit of fantasy, at least for the short term.

**Comment by terjeio** [comment]
> Why are there 2 3D View tabs? They seem redundant.

I can remove the top one - a remnant from before the app window could be resized.

> In Settings: Grbl. There is a large, empty light gray rectangle. I don't think it is ever used. Is there a purpose for it?

The one used for settings descriptions?

![image](https://user-images.githubusercontent.com/20260062/155831997-8035dbfe-6c2b-4a7d-b555-733616fbd6b3.png)

> I continue to think having a very simple Probe Z tab would be useful. 

Simple enough for a macro?

> I think it would be useful to have a list of error and alarm codes available from the app. Perhaps with some additional info to help the user resolve the problem. Maybe in Help?

I'll add that to todo.

> Same thing for a list of keyboard short cuts. Is the wiki usage tip list up to date?

I'll have to check that. At least the jogging shortcuts added for gamepad use are not in the list.

> On the subject of keyboard short cuts. Making them user assignable would be awesome. I realize that is a bit of fantasy, at least for the short term.

On my todo list already. The first step has been made by centralizing the keyboard input handling code.

**Reply by terjeio** [reply]
Keyboard shortcuts [can now be changed by the user](https://github.com/terjeio/ioSender/discussions/176#discussioncomment-2261131).

**Comment by phil-barrett** [comment]
>| In Settings: Grbl. There is a large, empty light gray rectangle. I don't think it is ever used. Is there a purpose for it?
> The one used for settings descriptions?

Odd, that does not show in 2.0.36p28  or 2.0.36 (release version).  In fact, I don't ever recall seeing anything in the rectangle. A problem with my machine?   It's late, will check tomorrow.

> <Z probing> Simple enough for a macro?

Sure, I've done that before.  Though for beginners, it may be a challenge. Are you suggesting it sits on the Macro Bar rather than a tab in  Probing?

**Reply by terjeio** [reply]
> Are you suggesting it sits on the Macro Bar rather than a tab in Probing?

Yes - simple means that no values are needed from the main part (left side) of the probing tab? If only a few are needed a flyout could be added instead? Or should I make a beginners version in addition to the XL version by removing some of the complexity/flexibility?

**Comment by phil-barrett** [comment]
> Odd, that does not show in 2.0.36p28 or 2.0.36 (release version). In fact, I don't ever recall seeing anything in the rectangle. A problem with my machine? It's late, will check tomorrow.

I've checked 3 machines here with fairly recent versions of ioSender and none of them are showing any settings descriptions.

**Comment by terjeio** [comment]
New [edge version up](https://github.com/terjeio/ioSender/discussions/176#discussioncomment-2257593).  I have added a menu item for listing alarm and error codes and have swapped the coolant and outline boxes. This hides the lower part of the spindle box at minimum window heigth though. I have tried to make use of all the vertical height seemingly available - but which is not due to the layout control I am using (the MDI and parser state controls takes up the whole window width at the bottom).
